src/test/scala/mesosphere/marathon/core/task/tracker/impl/TaskStatusUpdateProcessorImplTest.scala

```diff
@@ -9,118 +9,160 @@ import mesosphere.marathon.Protos.MarathonTask
 import mesosphere.marathon.core.CoreGuiceModule
 import mesosphere.marathon.core.base.ConstantClock
 import mesosphere.marathon.core.launchqueue.LaunchQueue
-import mesosphere.marathon.core.task.bus.TaskStatusObservables.TaskStatusUpdate
 import mesosphere.marathon.core.task.bus.{ TaskStatusEmitter, TaskStatusUpdateTestHelper }
-import mesosphere.marathon.core.task.tracker.TaskStatusUpdateProcessor
 import mesosphere.marathon.core.task.tracker.impl.steps._
 import mesosphere.marathon.event.MesosStatusUpdateEvent
 import mesosphere.marathon.health.HealthCheckManager
 import mesosphere.marathon.metrics.Metrics
 import mesosphere.marathon.state.{ AppDefinition, AppRepository, PathId, Timestamp }
 import mesosphere.marathon.tasks.{ TaskIdUtil, TaskTracker }
+import mesosphere.marathon.test.Mockito
 import mesosphere.marathon.{ MarathonSchedulerDriverHolder, MarathonSpec, MarathonTestHelper }
-import org.apache.mesos.Protos.TaskState
 import org.apache.mesos.SchedulerDriver
-import org.mockito.{ Matchers, ArgumentCaptor, Mockito }
+import org.mockito.ArgumentCaptor
+import org.scalatest.GivenWhenThen
+import org.scalatest.concurrent.ScalaFutures
 
 import scala.concurrent.duration._
 import scala.concurrent.{ Await, Future }
 
-class TaskStatusUpdateProcessorImplTest extends MarathonSpec {
+class TaskStatusUpdateProcessorImplTest extends MarathonSpec with Mockito with ScalaFutures with GivenWhenThen {
```

Comment:
:+1: for using ScalaFutures – there are still `Await.result` calls in the test, can you please replace them with `.futureValue`?